vmm/src/lib.rs

```diff
@@ -1036,15 +1050,19 @@ impl Vmm {
                                         break;
                                     }
                                 },
-                                Err(vstate::Error::VcpuRun(ref e)) => match e.errno() {
-                                    // Why do we check for these if we only return EINVAL?
-                                    libc::EAGAIN | libc::EINTR => {}
-                                    _ => {
-                                        METRICS.vcpu.failures.inc();
-                                        error!("Failure during vcpu run: {:?}", e);
-                                        break;
+                                // The unwrap on raw_os_error can only fail if we have a logic
+                                // error in our code in which case it is better to panic.
+                                Err(vstate::Error::VcpuRun(ref e)) => {
+                                    match e.raw_os_error().unwrap() {
+                                        // Why do we check for these if we only return EINVAL?
```

Comment:
I think this comment can go because the `ioctl` call itself might fail, in which case `errno` is set accordingly. Neither `EAGAIN` nor `EINTR` are irrecoverable so no reason to panic on them.